### PR TITLE
Enable  nvidia-smi command in AKS

### PR DESCRIPTION
**Pull request by Binyang2014**
Refer to issue: https://github.com/Azure/AKS/issues/1271
Copy nvidia-smi from /usr/local/nvidia/bin to /usr/local/bin will solve the problem.

### CODE REVIEW

File: src/cluster/config/cluster.py [hunk 1]
```diff
@@ -60,6 +60,8 @@ def validation_common(self, common_configuration):
             return False, "az-rdma is miss in service-configuration.yaml -> cluster -> common -> az-rdma"
         if "k8s-rbac" not in common_configuration:
             return False, "k8s-rbac is miss in service-configuration.yaml -> cluster -> common -> k8s-rbac"
+        if "deploy-in-aks" not in common_configuration:
+            return False, "deploy-in-aks is miss in service-configuration.yaml -> cluster ->common -> deploy-in-aks"
```
Comment (abuccts) [inline src/cluster/config/cluster.py]: `is miss` -> `is missing`

File: src/job-exporter/deploy/job-exporter.yaml.template [hunk 2]
```diff
@@ -102,6 +116,14 @@ spec:
         - name: gpu-config
           configMap:
             name: gpu-configuration
+        {%- if cluster_cfg['cluster']['common']['deploy-in-aks'] == "true" %}
+        - name: usr-bin
+          hostPath:
+            path: /usr/bin
+        - name: usr-local-nvidia-bin
+          hostPath:
+            path: /usr/local/nvidia/bin
```
Comment (abuccts) [inline src/job-exporter/deploy/job-exporter.yaml.template]: mount `/usr/local/nvidia`?

File: src/job-exporter/src/main.py [hunk 3]
```diff
@@ -54,7 +55,31 @@ def collect(self):
...
+    # Refer to issue: https://github.com/Azure/AKS/issues/1271, we need to set nvdia-smi manually
+    if deploy_env == "aks":
+        host_usr_bin_dir = os.environ.get("HOST_USR_BIN_DIR")
+        host_nvidia_bin_dir = os.environ.get("HOST_NVIDIA_BIN_DIR")
+        if not host_usr_bin_dir or not host_nvidia_bin_dir:
+            logger.error("Failed to get HOST_USR_BIN_DIR or HOST_NVIDIA_BIN_DIR env")
+            raise Exception("Environment not set correctly, HOST_NVIDIA_BIN_DIR is{},\
+                             HOST_NVIDIA_BIN_DIR is {}".format(host_usr_bin_dir, host_nvidia_bin_dir))
+
+        if os.path.isfile(host_usr_bin_dir + "/nvidia-smi"):
```
Comment (abuccts) [inline src/job-exporter/src/main.py]: use `os.path.join(_dir, 'nvidia-smi')`

File: src/job-exporter/deploy/job-exporter.yaml.template (hunk 2 above)
Comment (Binyang2014) [reply src/job-exporter/deploy/job-exporter.yaml.template]: Fixed

---
In reply to: [336347682](https://github.com/microsoft/pai/pull/3749#discussion_r336347682) [](ancestors = 336347682)

File: src/job-exporter/src/main.py (hunk 3 above)
Comment (Binyang2014) [reply src/job-exporter/src/main.py]: Fixed

---
In reply to: [336347774](https://github.com/microsoft/pai/pull/3749#discussion_r336347774) [](ancestors = 336347774)

**Pull request merged**